processing.go

```diff
@@ -81,14 +89,18 @@ func grabTarget(input ScanTarget, m *Monitor) []byte {
 	moduleResult := make(map[string]ScanResponse)
 
 	for _, scannerName := range orderedScanners {
+		scanner := scanners[scannerName]
+		trigger := (*scanner).GetTrigger()
+		if (input.Tag != "" || trigger != "") && input.Tag != trigger {
```

Comment:
When looking over this last night, I read this as `if (input.Tag != "" && trigger != "") && input.Tag != trigger`, which seems more like the behavior I would expect -- namely, no-tag-matches-everything (bolstered by the observation that as it stands, the first clause is moot -- if it is false, i.e. both are empty, then the second clause is false).

This change would allow you to reuse tagged input with un-tagged scans, and probably more importantly, tagged scans with un-tagged input.

It would mean that if you re-use untagged input with a tagged scanner, all modules (rather than just untagged modules) would hit every host -- but in that case, suffixing each input line with a dummy tag would be a trivial workaround to get this behavior back.

Reply:
Thanks for the thoughtful review, Justin.

I considered no-tag-matches-everything, but that seems unsafe as a default behavior.  Suppose someone accidentally omits the trigger when adding a new scanner to a configuration with dozens of tags.  They'll make boatloads of spurious connections, and might not notice unless they're paying attention to the failure rate.  With strict-tag-equals-trigger, the error is obvious immediately, since the scan never triggers.

What do others think?

Reply:
Amended PR to fix the moot clause in the conditional.  Good catch Justin!

Reply:
That makes sense -- basically, if you are using triggers, it is preferable for a misconfigured scanner to scan no hosts than for it to scan hosts that it was not intended to scan.